src/index.js

```diff
@@ -1,81 +1,171 @@
-      </button>
-    );
-  }
-}
-
-class Panel extends React.Component {
+class Calculator extends Component {
   constructor(props) {
     super(props);
 
     this.state = {
       input: "",
       previousNumber: "",
-      currentNumber: "",
       operator: "",
+      nextOperation: "",
     };
   }
 
-  handleClick(i) {
-    console.log("value i= " + i);
-  }
+  addToInput = (val) => {
+    if (this.state.nextOperation !== "clear")
+      this.setState({ input: this.state.input + val });
+    else
+      this.setState({
+        input: val,
+        nextOperation: "",
+      });
+  };
 
-  renderButton(i) {
-    return <Button name={i} onClick={() => this.handleClick(i)} />;
-  }
+  add = () => {
+    this.setState({
+      previousNumber: this.state.input,
+      input: "",
+      operator: "plus",
+    });
+  };
 
-  render() {
-    return (
-      <div>
-        <div id="display" className="display-row"></div>
-        <div className="board-row">
-          {this.renderButton("AC")}
-          {this.renderButton("+/-")}
-          {this.renderButton("%")}
-          {this.renderButton("÷")}
-        </div>
-        <div className="board-row">
-          {this.renderButton("1")}
-          {this.renderButton("2")}
-          {this.renderButton("3")}
-          {this.renderButton("*")}
-        </div>
-        <div className="board-row">
-          {this.renderButton("4")}
-          {this.renderButton("5")}
-          {this.renderButton("6")}
-          {this.renderButton("-")}
-        </div>
-        <div className="board-row">
-          {this.renderButton("7")}
-          {this.renderButton("8")}
-          {this.renderButton("9")}
-          {this.renderButton("+")}
-        </div>
-        <div className="board-row">
-          {this.renderButton("00")}
-          {this.renderButton("0")}
-          {this.renderButton(".")}
-          {this.renderButton("=")}
-        </div>
-      </div>
-    );
-  }
-}
+  divide = () => {
+    this.setState({
+      previousNumber: this.state.input,
+      input: "",
+      operator: "divide",
+    });
+  };
+
+  multiply = () => {
+    this.setState({
+      previousNumber: this.state.input,
+      input: "",
+      operator: "multiply",
+    });
+  };
+
+  subtract = () => {
+    this.setState({
+      previousNumber: this.state.input,
+      input: "",
+      operator: "subtract",
+    });
+  };
+
+  percent = () => {
+    let result = this.state.previousNumber / 100;
+    this.setState({
+      input: result,
+    });
+  };
+
+  addDecimal = (val) => {
+    if (this.state.input.indexOf(".") === -1)
+      this.setState({ input: this.state.input + val });
+  };
+
+  cutInput = () => {
+    this.setState({
+      input: this.state.input.slice(0, -1),
+    });
+  };
+
+  clearInput = () => {
+    this.setState({
+      input: "",
+      previousNumber: "",
+      operator: "",
+    });
+  };
+
+  evaluteInput = () => {
+    if (this.state.operator === "plus") {
+      let result =

```

Comment:
Let's move this code to little functions e.g., `addInput()`, `subtractInput()` etc..